gax-httpjson/src/main/java/com/google/api/gax/httpjson/EmptyMessage.java

```diff
@@ -58,4 +59,62 @@ public Object getFieldValue(String fieldName) {
   public ApiMessage getApiMessageRequestBody() {
     return null;
   }
+
+  @Override
+  public boolean equals(Object other) {
+    return other instanceof EmptyMessage;
+  }
+
+  private static final EmptyMessage DEFAULT_INSTANCE;
+
+  public static EmptyMessage getDefaultInstance() {
+    return DEFAULT_INSTANCE;
+  }
+
+  private EmptyMessage() {}
+
+  public static Builder newBuilder() {
+    return DEFAULT_INSTANCE.toBuilder();
+  }
+
+  public static Builder newBuilder(EmptyMessage prototype) {
+    return DEFAULT_INSTANCE.toBuilder().mergeFrom(prototype);
+  }
+
+  public Builder toBuilder() {
+    return this == DEFAULT_INSTANCE ? new Builder() : new Builder().mergeFrom(this);
+  }
+
+  static {
+    DEFAULT_INSTANCE = new EmptyMessage();
+  }
+
+  public static class Builder {
+    Builder() {}
+
+    public Builder mergeFrom(EmptyMessage other) {
+      return this;
+    }
+
+    public Builder(EmptyMessage other) {}
+
+    public EmptyMessage build() {
+      return new EmptyMessage();
+    }
+
+    public Builder clone() {
+      Builder newBuilder = new Builder();
```

Comment:
We could just do `return new Builder();`?
OTOH, is the method really needed though? cloning a `Builder` object doesn't sound like a valid use case.

Reply:
This just follows the protobuf java paradigm: https://github.com/googleapis/google-cloud-java/blob/c011130c0114ccdcde9bda60202f334722df331d/google-api-grpc/proto-google-cloud-dataproc-v1/src/main/java/com/google/cloud/dataproc/v1/Cluster.java#L2595

Reply:
Because com.google.protobuf.Empty is also a GeneratedMessageV3, I'm pretty sure it would also look like this. (idk how to look at generated protobuf classes tho so i can't confirm)

Reply:
IntelliJ usually downloads the sources of the maven artifacts used in project automatically, so you can navigate to the sources just from there. If it can't find sources it will show decompiled version (can be good enough as well). The sources of maven artifacts are usually published together with compiled classes (sometimes in same jar, sometimes as a separate `-sources.jar`). For the `Empty` class, its sources can be found here: [protobuf-java-3.6.1](http://central.maven.org/maven2/com/google/protobuf/protobuf-java/3.6.1/protobuf-java-3.6.1-sources.jar)

Reply:
If this method should be there, let's just keep the method and change it to `return new Builder();` then :)

Reply:
Done.